File: Java/2825. Make String a Subsequence Using Cyclic Increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int i1 = 0, i2 = 0, s1 = str1.length(), s2 = str2.length();
        while (i1 < s1 && i2 < s2) {
            char c1 = str1.charAt(i1), c2 = str2.charAt(i2);
            if (c1 == c2 || ((c1 - 'a' + 1) % 26) == (c2 - 'a')) {
                i2++;
            }
            i1++;
        }
        return i2 == s2;
    }
}

File: Python3/2825. Make String a Subsequence Using Cyclic Increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        if len(str1) < len(str2):
            return False

        a = ord('a')

        i1 = i2 = 0
        while i1 < len(str1) and i2 < len(str2):
            c1, c2 = str1[i1], str2[i2]
            if c1 == c2 or (ord(c1) + 1 - a) % 26 == ord(c2) - a:
                i2 += 1
            i1 += 1

        return i2 == len(str2)